blender_3dmigoto.py

```diff
@@ -1567,32 +1572,29 @@ def write_ini_file(f, vb, vb_path, ib, ib_path, strides, obj, topology):
             backup_section += 'ResourceBakVB{0} = ref vb{0}\n'.format(vbuf_idx or 0)
             restore_section += 'vb{0} = ResourceBakVB{0}\n'.format(vbuf_idx or 0)
 
-    # FIXME: Maybe split this into several ini files that the user may or may
-    # not choose to generate? One that just lists resources, a second that
-    # lists the TextureOverrides to replace draw calls, and a third with the
-    # ShaderOverride sections (or a ShaderRegex for foolproof replacements)...?
-    f.write(textwrap.dedent('''
-            ; Automatically generated file, be careful not to overwrite if you
-            ; make any manual changes
-
-            ; Please note - it is not recommended to place the [ShaderOverride]
-            ; here, as you only want checktextureoverride executed once per
-            ; draw call, so it's better to have all the shaders listed in a
-            ; common file instead to avoid doubling up and to allow common code
-            ; to enable/disable the mods, backup/restore buffers, etc. Plus you
-            ; may need to locate additional shaders to take care of shadows or
-            ; other render passes. But if you understand what you are doing and
-            ; need a quick 'n' dirty way to enable the reinjection, fill this in
-            ; and uncomment it:
-            ;[ShaderOverride{suffix}]
-            ;hash = FILL ME IN...
-            ;checktextureoverride = vb0
-
-            [TextureOverride{suffix}]
-            ;hash = FILL ME IN...
-            ''').lstrip().format(
-                suffix='',
-            ))
+    #Write ShaderOverrides file
+    shader_overrides_path = os.path.dirname(os.path.dirname(os.path.dirname(f.name))) + r'\ShaderOverrides'
+    if not os.path.exists(shader_overrides_path):
+        os.makedirs(shader_overrides_path)
+
+    if override_vertex_shader:
+        shader_hash_to_write = obj['3DMigoto:VSHash']
+    else:
+        shader_hash_to_write = obj['3DMigoto:PSHash']
+        
+    shader_overrides_file = shader_overrides_path + '\\' + shader_hash_to_write + '.ini'
+    check_texture_override_n = 4
+    
+    with open(shader_overrides_file, 'w') as f_overrides:
+        f_overrides.write('[ShaderOverride]\nhash = {}\nchecktextureoverride = vb0\n'.format(shader_hash_to_write))
```

Comment:
I'm not a fan of putting newlines as inline '\n' - for these multiline strings I'd generally prefer to use ''' to directly write the string in multiline directly into the scrtipt, using helpers like textwrap.dedent() to remove common indentations, and lstrip to remove any extra whitespace resulting from an initial newline.

Same comment to the below lines.

Reply:
ok, fixed